sdk/identity/Azure.Identity/src/CredentialDiagnosticScope.cs

```diff
@@ -14,51 +14,69 @@ namespace Azure.Identity
         private readonly string _name;
         private readonly DiagnosticScope _scope;
         private readonly TokenRequestContext _context;
+        private readonly IScopeHandler _scopeHandler;
 
-        public CredentialDiagnosticScope(string name, DiagnosticScope scope, TokenRequestContext context)
+        public CredentialDiagnosticScope(string name, TokenRequestContext context, IScopeHandler scopeHandler)
         {
             _name = name;
-
-            _scope = scope;
-
+            _scope = scopeHandler.CreateScope(name);
             _context = context;
+            _scopeHandler = scopeHandler;
         }
 
         public void Start()
         {
-            _scope.Start();
+            AzureIdentityEventSource.Singleton.GetToken(_name, _context);
```

Comment:
`AzureIdentityEventSource.Singleton.GetToken(_name, _context)` call is moved here from `CredentialPipeline.StartGetTokenScope`.